flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainerTestEnvironment.java

```diff
@@ -27,61 +27,74 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
+import java.util.Collection;
 import java.util.stream.Collectors;
 
-import static org.apache.flink.configuration.HeartbeatManagerOptions.HEARTBEAT_INTERVAL;
-import static org.apache.flink.configuration.HeartbeatManagerOptions.HEARTBEAT_TIMEOUT;
-import static org.apache.flink.configuration.JobManagerOptions.SLOT_REQUEST_TIMEOUT;
-import static org.apache.flink.configuration.MetricOptions.METRIC_FETCHER_UPDATE_INTERVAL;
-import static org.apache.flink.configuration.TaskManagerOptions.NUM_TASK_SLOTS;
-import static org.apache.flink.connector.testframe.utils.ConnectorTestConstants.HEARTBEAT_INTERVAL_MS;
-import static org.apache.flink.connector.testframe.utils.ConnectorTestConstants.HEARTBEAT_TIMEOUT_MS;
-import static org.apache.flink.connector.testframe.utils.ConnectorTestConstants.METRIC_FETCHER_UPDATE_INTERVAL_MS;
-import static org.apache.flink.connector.testframe.utils.ConnectorTestConstants.SLOT_REQUEST_TIMEOUT_MS;
-
 /** Test environment running job on {@link FlinkContainers}. */
 public class FlinkContainerTestEnvironment implements TestEnvironment, ClusterControllable {
 
     private static final Logger LOG = LoggerFactory.getLogger(FlinkContainerTestEnvironment.class);
 
     private final FlinkContainers flinkContainers;
-    private final List<String> jarPaths = new ArrayList<>();
+    private final Collection<String> jarPaths = new ArrayList<>();
+    private String checkpointPath = FlinkContainersConfig.getDefaultCheckpointPath();
 
     public FlinkContainerTestEnvironment(
             int numTaskManagers, int numSlotsPerTaskManager, String... jarPaths) {
-        this(new Configuration(), numTaskManagers, numSlotsPerTaskManager, jarPaths);
+        this(new Configuration(), numTaskManagers, numSlotsPerTaskManager, Arrays.asList(jarPaths));
     }
 
     public FlinkContainerTestEnvironment(
             Configuration clusterConfiguration,
             int numTaskManagers,
             int numSlotsPerTaskManager,
             String... jarPaths) {
-        Configuration config = new Configuration();
-        config.addAll(clusterConfiguration);
-        config.set(NUM_TASK_SLOTS, numSlotsPerTaskManager);
-        config.set(HEARTBEAT_INTERVAL, HEARTBEAT_INTERVAL_MS);
-        config.set(HEARTBEAT_TIMEOUT, HEARTBEAT_TIMEOUT_MS);
-        config.set(SLOT_REQUEST_TIMEOUT, SLOT_REQUEST_TIMEOUT_MS);
-        config.set(METRIC_FETCHER_UPDATE_INTERVAL, METRIC_FETCHER_UPDATE_INTERVAL_MS);
+        this(
+                clusterConfiguration,
+                numTaskManagers,
+                numSlotsPerTaskManager,
+                Arrays.asList(jarPaths));
+    }
+
+    public FlinkContainerTestEnvironment(
+            Configuration clusterConfiguration,
+            int numTaskManagers,
+            int numSlotsPerTaskManager,
+            Collection<String> jarPaths) {
```

Comment:
Why do we need the Collection vs the variable length String arg?

Reply:
You cannot just forward varargs to another constructor that has varargs, since it is converted to an Array implicitly. I prefer working with collections not arrays in this case.